client.go

```diff
@@ -753,3 +765,15 @@ func (s *Client) Probe(ctx context.Context, req *ProbeRequest) error {
 		req.RouteHints,
 	)
 }
+
+// AbandonSwap sends the hash of the swap to abandon to all active swaps. Only
+// the addressed swap will abandon itself.
+func (s *Client) AbandonSwap(req *AbandonSwapRequest) error {
+	if req == nil {
+		return errors.New("no request provided")
+	}
+
+	s.abandonChans[req.SwapHash] <- struct{}{}
```

Comment:
According to my previous comment we could make use of `ctx` here and do smth like this 
```suggestion
	select {
	case s.abandonChans[req.SwapHash] <- struct{}{}:
	case ctx.Done():
	    return ctx.Err()
	}
```
in order to avoid blocking indefinitely.

This also raises a new topic: How are channels erased from memory?
And I think an important parameter here is whether the channel is buffered or not (i.e whether you block while writing to it or not)

1. Buffered channel: That means you immediately return from the above line thus there is also no need for using `ctx`
    Note: How/When could we delete entries from the map in this case?
2. Unbuffered channel: That means we block in the above line and when the execution resumes either
    i) ctx is done
    ii) swap actually read the signal
    and since the channel is only used once we could safely then delete the channel from the map

Seems to me like sticking with 2 (unbuffered channels) is more simple, so I would suggest adding this line on top of the function
```golang
// When this function returns there is no use of the swap's abandon chan.
defer delete(s.abandonChans, req.SwapHash)
```
given that the upper block of code with the `select` statement and `ctx` is used too

Reply:
Or if there is another place in the code where `client` receives signals from swaps when they reach a final state we could delete the map entry there, as it is no longer of use

Reply:
Great points @GeorgeTsagk, thank you!

I think the `defer delete` however should only be applied if the channel was successfully abandoned, which could still fail even if it was sent on abandon channel.

Let me check out the places where we delete from the map.

Thanks again.

Reply:
The deletion introduces another problem, namely concurrent access to the channel map. I solved this by introducing a Mutx to the `executor` that is locked when a loop-in is created and locked again when a swap finishes.

Would be interested in alternative solutions.